packages/ra-core/src/controller/CreateController.tsx

```diff
@@ -83,95 +94,57 @@ interface EnhancedProps {
  *     );
  *     export default App;
  */
-export class UnconnectedCreateController extends Component<
-    Props & EnhancedProps
-> {
-    public static defaultProps: Partial<Props> = {
-        record: {},
-    };
+const CreateController = (props: Props) => {
+    useCheckMinimumRequiredProps(
+        'Create',
+        ['basePath', 'location', 'resource'],
+        props
+    );
+    const {
+        basePath,
+        children,
+        resource,
+        location,
+        record = {},
+        hasShow,
+        hasEdit,
+    } = props;
 
-    private record: Partial<Record>;
+    const translate = useTranslate();
+    const dispatch = useDispatch();
+    const recordToUse = getRecord(location, record);
+    const isLoading = useSelector(
+        (state: ReduxState) => state.admin.loading > 0
+    );
 
-    constructor(props) {
-        super(props);
-        const {
-            location: { state, search },
-            record,
-        } = this.props;
-        this.record =
```

Comment:
I think you've broken the clone feature by not porting this

Reply:
The e2e tests disagree with you ;)

Reply:
And btw I ported this. This is the `getRecord` function you commented earlier